Simulation crashes when restarting with changed GPU layout

I started a simulation of the LaserWakefield example from the `1.cfg` (1x1x1 GPU) and created a single checkpoint. Then I restarted the simulation from this checkpoint with a modified GPU layout. The simulation loads from the checkpoint and finishes the initialization but crashes shortly after starting.
Not all layouts crash but most do. See below for a list. 

## Setup

PIConGPU commit 1c83e01bdf75c89a241c67e1690671b29c5642a1 (newest dev at time of writing)
LaserWakefield example with periodic boundaries turned off (due to #5675)
build command: `pic-build -c "-DPIC_VERBOSE=127 -DPMACC_VERBOSE=127 -DPMACC_BLOCKING_KERNEL=ON -DPMACC_ASYNC_QUEUES=OFF"`
Tested on Jupiter Booster

### GPU layouts tested
Restarting from:
1x2x1 and 2x2x1 do **not** crash.
2x4x1 (corresponds to standard `8.cgf`), 1x4x2, 2x2x2 and 1x2x2 did crash.

Of note is that 1x2x2 crashes but 2x2x1 does not, so there seems to be an asymmetry here. 
I tested both multiple times to make sure that the crashes are not just random.

## Logs  (from the 2x4x1 restart)
### `stderr`
```
Die folgenden Module wurden nicht entladen:
  (Benutzen Sie "module --force purge" um alle Module zu entladen):

  1) Stages/2026
ln: die symbolische Verknüpfung 'output' konnte nicht angelegt werden: Die Datei existiert bereits
[alpaka] Error: </e/project1/pwfa-trojan/wrobel1/src/picongpu/include/picongpu/../pmacc/exec/KernelLauncher.tpp>:114 /e/project1/pwfa-trojan/wrobel1/src/picongpu/thirdParty/alpaka/include/alpaka/dev/DevUniformCudaHipRt.hpp(257) 'TApi::setDevice(dev.getNativeHandle())' A previous API call (not this one) set the error  : 'cudaErrorInvalidValue': 'invalid argument'!
Crash after kernel launch N5pmacc8lockstep4exec6detail14LockStepKernelINS_9particles10operations6detail27KernelSplitIntoListOfFramesENS0_8BlockCfgINS_4math2CT6VectorISt17integral_constantIjLj256EENSA_6detail2naESF_EEEEEE [/e/project1/pwfa-trojan/wrobel1/src/picongpu/include/picongpu/../pmacc/particles/operations/splitIntoListOfFrames.kernel:303 ]
[alpaka] Error: </e/project1/pwfa-trojan/wrobel1/src/picongpu/include/picongpu/../pmacc/exec/KernelLauncher.tpp>:114 /e/project1/pwfa-trojan/wrobel1/src/picongpu/thirdParty/alpaka/include/alpaka/dev/DevUniformCudaHipRt.hpp(257) 'TApi::setDevice(dev.getNativeHandle())' A previous API call (not this one) set the error  : 'cudaErrorInvalidValue': 'invalid argument'!
Crash after kernel launch N5pmacc8lockstep4exec6detail14LockStepKernelINS_9particles10operations6detail27KernelSplitIntoListOfFramesENS0_8BlockCfgINS_4math2CT6VectorISt17integral_constantIjLj256EENSA_6detail2naESF_EEEEEE [/e/project1/pwfa-trojan/wrobel1/src/picongpu/include/picongpu/../pmacc/particles/operations/splitIntoListOfFrames.kernel:303 ]
.
. // 49 more times the same error
.
[alpaka] Error: </e/project1/pwfa-trojan/wrobel1/src/picongpu/include/picongpu/../pmacc/exec/KernelLauncher.tpp>:114 /e/project1/pwfa-trojan/wrobel1/src/picongpu/thirdParty/alpaka/include/alpaka/dev/DevUniformCudaHipRt.hpp(257) 'TApi::setDevice(dev.getNativeHandle())' A previous API call (not this one) set the error  : 'cudaErrorIllegalAddress': 'an illegal memory access was encountered'!
Crash after kernel launch N5pmacc8lockstep4exec6detail14LockStepKernelIN8picongpu26KernelMoveAndMarkParticlesINS_20SuperCellDescriptionINS_4math2CT6VectorISt17integral_constantIiLi4EESA_IiLi8EESC_EENS9_ISA_IiLi1EESE_SE_EENS9_ISA_IiLi2EESG_SG_EEEEEENS0_8BlockCfgINS9_ISA_IjLj256EENS8_6detail2naESN_EEEEEE [/e/project1/pwfa-trojan/wrobel1/src/picongpu/include/picongpu/../picongpu/particles/Particles.tpp:356 ]
/e/project1/pwfa-trojan/wrobel1/src/picongpu/thirdParty/alpaka/include/alpaka/mem/buf/uniformCudaHip/traits/BufUniformCudaHipRtTraits.hpp(251) 'TApi::free(ptr)' returned error  : 'cudaErrorIllegalAddress': 'an illegal memory access was encountered'!
terminate called after throwing an instance of 'std::runtime_error'
  what():  /e/project1/pwfa-trojan/wrobel1/src/picongpu/thirdParty/alpaka/include/alpaka/event/EventUniformCudaHipRt.hpp(143) 'ret = TApi::eventQuery(event.getNativeHandle())' returned error  : 'cudaErrorIllegalAddress': 'an illegal memory access was encountered'!
srun: error: jpbo-029-15: task 5: Aborted (core dumped)
slurmstepd: error:  mpi/pmix_v5: _errhandler: jpbo-029-15 [1]: pmixp_client_v2.c:211: Error handler invoked: status = -61, source = [slurm.pmix.437146.0:5]
srun: Job step aborted: Waiting up to 32 seconds for job step to finish.
slurmstepd: error: *** STEP 437146.0 ON jpbo-029-14 CANCELLED AT 2026-05-12T09:19:06 ***
srun: error: jpbo-029-15: tasks 4,6-7: Killed
srun: error: jpbo-029-14: tasks 0-3: Killed
```

### `stdout`
see file [stdout](https://github.com/user-attachments/files/27625758/stdout.txt)

--- a/include/picongpu/plugins/openPMD/restart/LoadSpecies.hpp
+++ b/include/picongpu/plugins/openPMD/restart/LoadSpecies.hpp
@@ -114,8 +114,11 @@ namespace picongpu
                                 char filterCurrent = filterKeep;
                                 for(size_t d = 0; d < simDim; ++d)
                                 {
-                                    auto positionInD = positionVec[d] + positionOffsetVec[d];
-                                    if(positionInD < patchTotalOffset[d] || positionInD >= patchUpperCorner[d])
+                                    signed long long p = positionVec[d];
+                                    signed long long o = positionOffsetVec[d];
+                                    signed long long patch_bottom = patchTotalOffset[d];
+                                    signed long long patch_up = patchUpperCorner[d];
+                                    if(p < patch_bottom - o || p >= patch_up - o)
                                     {
                                         filterCurrent = filterRemove;
                                         break;

--- a/include/picongpu/plugins/openPMD/restart/LoadSpecies.hpp
+++ b/include/picongpu/plugins/openPMD/restart/LoadSpecies.hpp
@@ -114,7 +114,7 @@ namespace picongpu
                                 char filterCurrent = filterKeep;
                                 for(size_t d = 0; d < simDim; ++d)
                                 {
-                                    signed long long p = positionVec[d];
+                                    auto p = positionVec[d];
                                     signed long long o = positionOffsetVec[d];
                                     signed long long patch_bottom = patchTotalOffset[d];
                                     signed long long patch_up = patchUpperCorner[d];

--- a/include/picongpu/plugins/openPMD/restart/LoadSpecies.hpp
+++ b/include/picongpu/plugins/openPMD/restart/LoadSpecies.hpp
@@ -347,15 +347,18 @@ namespace picongpu
                                 "openPMD: Keeping %1% of the current batch's %2% particles after filtering.")
                                 % numParticlesAfterFiltering % numParticlesCurrentBatch;
 
-                            pmacc::particles::operations::splitIntoListOfFrames(
-                                *speciesTmp,
-                                mappedFrame,
-                                // @attention not numParticlesCurrentBatch, filtered vs. unfiltered number
-                                numParticlesAfterFiltering,
-                                cellOffsetToTotalDomain,
-                                totalCellIdx_,
-                                *threadParams->cellDescription,
-                                picLog::INPUT_OUTPUT());
+                            if(numParticlesAfterFiltering > 0)
+                            {
+                                pmacc::particles::operations::splitIntoListOfFrames(
+                                    *speciesTmp,
+                                    mappedFrame,
+                                    // @attention not numParticlesCurrentBatch, filtered vs. unfiltered number
+                                    numParticlesAfterFiltering,
+                                    cellOffsetToTotalDomain,
+                                    totalCellIdx_,
+                                    *threadParams->cellDescription,
+                                    picLog::INPUT_OUTPUT());
+                            }
                         });
                 }
             };

